pkg/k8s/watchers/cilium_endpoint.go

```diff
@@ -89,6 +89,10 @@ func (k *K8sWatcher) ciliumEndpointsInit(ciliumNPClient *k8s.K8sCiliumClient, as
 		k.blockWaitGroupToSyncResources(isConnected, nil, ciliumEndpointInformer, k8sAPIGroupCiliumEndpointV2)
 
 		once.Do(func() {
+			if err := k8s.WaitForCRD(k8s.APIExtClient(), cilium_v2.CEPName); err != nil {
+				log.WithError(err).Fatalf("Unable to find %s CRD", cilium_v2.CEPName)
+			}
```

Comment:
These wait operations are blocking, before each watcher. Should we consider trying to move the waiting mechanism so that it's not blocking and parallelize it?

Reply:
The `WaitForCRD` creates a watcher for each call, which we want to avoid. Instead, we can reuse the `k.blockWaitGroupToSyncResources` and in here we can have `k.WaitForCacheSync("CRD"+cilium_v2.CEPName)`